site/app/controllers/admin/GradeOverrideController.php

```diff
@@ -54,11 +88,45 @@ public function updateOverriddenGrades($gradeable_id) {
         }
 
         if (((!isset($_POST['marks'])) || $_POST['marks'] == "" || is_float($_POST['marks']))) {
-            $error = "Marks be a integer";
+            $error = "Marks must be an integer";
             return $this->core->getOutput()->renderJsonFail($error);
         }
 
-        $this->core->getQueries()->updateGradeOverride($_POST['user_id'], $gradeable_id, $_POST['marks'], $_POST['comment']);
-        $this->getOverriddenGrades($gradeable_id);
+        $team = $this->core->getQueries()->getTeamByGradeableAndUser($gradeable_id, $_POST['user_id']);
+        //0 is for single submission, 1 is for team submission
+        $option = isset($_POST['option']) ? $_POST['option'] : -1;
+        if ($team != null && $team->getSize() > 1) {
+            if ($option == 0) {
+                $this->core->getQueries()->updateGradeOverride($_POST['user_id'], $gradeable_id, $_POST['marks'], $_POST['comment']);
+                return $this->getOverriddenGrades($gradeable_id);
```

Comment:
This seems to be pretty similar to the block of code above.  Would it be productive to share code between these two code blocks?

Reply:
It definitely would be but I wouldn't know how to approach it in PHP, it would also be nice to share the code with the extensions page if we can.